pkgs/development/libraries/gettext-runtime/default.nix

```diff
@@ -1,31 +1,29 @@
-{ stdenv, lib, fetchurl, libiconv, xz, fetchpatch }:
+{ stdenv, lib, fetchurl, libiconv }:
 
-stdenv.mkDerivation rec {
-  pname = "gettext";
-  version = "0.20.1";
+let common = import ../gettext/common.nix { inherit fetchurl; };
+in stdenv.mkDerivation rec {
+  pname = "gettext-runtime";
+  inherit (common) version src;
 
-  src = fetchurl {
-    url = "mirror://gnu/gettext/${pname}-${version}.tar.gz";
-    sha256 = "0p3zwkk27wm2m2ccfqm57nj7vqkmfpn7ja1nf65zmhz8qqs5chb6";
-  };
+  sourceRoot = "${common.pname}-${common.version}/${pname}/";
+
+  # Sources are within a subdir of the upstream directory. Given
+  # that patches are applied from sourceRoot, we cannot just
+  # fetch patches upstream in order to backport them. We have
+  # to update them to accomodate the effective root.
   patches = [
     ./absolute-paths.diff
-    ./gettext.git-2336451ed68d91ff4b5ae1acbc1eca30e47a86a9.patch
-  ]
-  ++ lib.optional stdenv.isDarwin
-      (fetchpatch {
-        url = "https://git.savannah.gnu.org/cgit/gettext.git/patch?id=ec0e6b307456ceab352669ae6bccca9702108753";
-        sha256 = "0xqs01c7xl7vmw6bqvsmrzxxjxk2a4spcdpmlwm3b4hi2wc2lxnf";
-      });
+  ] ++ lib.optional stdenv.isDarwin [
+    ./gettext.git-ec0e6b307456ceab352669ae6bccca9702108753.patch
+  ];
 
   outputs = [ "out" "man" "doc" "info" ];
```

Comment:
Could you send tree of each package's output?

Maybe also `lib`

Reply:
```
$ nix-build -A gettext-runtime.out -A gettext-runtime.man -A gettext-runtime.doc -A gettext-runtime.info|xargs tree
/nix/store/akia4d9zd8lmh0dfqzh81zccmrr115y8-gettext-runtime-0.20.1
├── bin
│   ├── envsubst
│   ├── gettext
│   ├── gettext.sh
│   └── ngettext
├── include
│   └── autosprintf.h
├── lib
│   ├── libasprintf.la
│   ├── libasprintf.so -> libasprintf.so.0.0.0
│   ├── libasprintf.so.0 -> libasprintf.so.0.0.0
│   └── libasprintf.so.0.0.0
├── nix-support
│   └── setup-hook
└── share
    ├── gettext
    │   └── ABOUT-NLS
    └── locale
        ├── ast
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── be
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── bg
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── ca
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── cs
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── da
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── de
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── el
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── en@boldquot
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── en@quot
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── eo
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── es
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── et
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── fi
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── fr
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── ga
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── gl
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── hr
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── hu
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── id
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── it
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── ja
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── ko
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── nb
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── nl
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── nn
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── pl
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── pt
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── pt_BR
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── ro
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── ru
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── sk
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── sl
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── sr
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── sv
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── tr
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── uk
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── vi
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── zh_CN
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        ├── zh_HK
        │   └── LC_MESSAGES
        │       └── gettext-runtime.mo
        └── zh_TW
            └── LC_MESSAGES
                └── gettext-runtime.mo
/nix/store/115fa7117bv4w2sd9ysbwmrj9k9w5z3c-gettext-runtime-0.20.1-man
└── share
    └── man
        ├── man1
        │   ├── envsubst.1.gz
        │   ├── gettext.1.gz
        │   └── ngettext.1.gz
        └── man3
            ├── bindtextdomain.3.gz
            ├── bind_textdomain_codeset.3.gz
            ├── dcgettext.3.gz
            ├── dcngettext.3.gz
            ├── dgettext.3.gz
            ├── dngettext.3.gz
            ├── gettext.3.gz
            ├── ngettext.3.gz
            └── textdomain.3.gz
/nix/store/b3hyj7c3bpk33rbglb451x6kspsbdwfp-gettext-runtime-0.20.1-doc
└── share
    └── doc
        └── gettext-runtime
            ├── autosprintf_all.html
            ├── bindtextdomain.3.html
            ├── bind_textdomain_codeset.3.html
            ├── csharpdoc
            │   ├── begin.html
            │   ├── GNU_Gettext_GettextResourceManager.html
            │   ├── GNU_Gettext_GettextResourceSet.html
            │   ├── GNU_Gettext.html
            │   ├── index.html
            │   └── namespaces.html
            ├── envsubst.1.html
            ├── gettext.1.html
            ├── gettext.3.html
            ├── javadoc2
            │   ├── allclasses-frame.html
            │   ├── deprecated-list.html
            │   ├── gnu
            │   │   └── gettext
            │   │       ├── GettextResource.html
            │   │       ├── package-frame.html
            │   │       ├── package-summary.html
            │   │       └── package-tree.html
            │   ├── help-doc.html
            │   ├── index-all.html
            │   ├── index.html
            │   ├── overview-tree.html
            │   ├── package-list
            │   ├── packages.html
            │   ├── serialized-form.html
            │   └── stylesheet.css
            ├── ngettext.1.html
            ├── ngettext.3.html
            └── textdomain.3.html
/nix/store/4d8pjvwshd41s96kjngmd89j7scpnpyx-gettext-runtime-0.20.1-info
└── share
    └── info
        └── autosprintf.info

102 directories, 94 files
```